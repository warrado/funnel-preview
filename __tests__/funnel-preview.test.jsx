import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";

import FunnelPreview from "../app/components/funnel-preview";
import PreviewPage from "../app/components/preview-page";

describe("FunnelPreview", () => {
  it("renders", () => {
    const { container } = render(
      <FunnelPreview bgColor="white">
        <PreviewPage></PreviewPage>
        <PreviewPage></PreviewPage>
      </FunnelPreview>
    );
    expect(container).toMatchSnapshot();
  });

  it("renders msg when no children", () => {
    const { container } = render(
      <FunnelPreview bgColor="white">
      </FunnelPreview>
    );

    const textElement = screen.queryByText("No funnel data to show.", {exact: false});
    expect(textElement).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
