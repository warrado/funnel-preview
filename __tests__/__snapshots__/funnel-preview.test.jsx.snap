// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FunnelPreview renders 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-items-center"
  >
    <div
      class="bg-cover bg-center h-[919px] w-[453px] pt-[96px] pb-[103px] pl-[26px] pr-[20px] z-1"
      style="background-image: url(/smartphone-cropped.png);"
    >
      <div
        class="h-full w-full p-4"
        style="background-color: white;"
      >
        <div
          class="slick-slider slick-initialized"
          dir="ltr"
        >
          <div
            class="slick-list"
          >
            <div
              class="slick-track"
              style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-hidden="false"
                class="slick-slide slick-active slick-current"
                data-index="0"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="overflow-auto w-full h-full text-center"
                  />
                </div>
              </div>
              <div
                aria-hidden="true"
                class="slick-slide"
                data-index="1"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="overflow-auto w-full h-full text-center"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="grid grid-cols-2 justify-items-center flex-none w-full text-center mt-4"
    >
      <svg
        aria-hidden="true"
        class="size-12 text-blue-700"
        data-slot="icon"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m11.25 9-3 3m0 0 3 3m-3-3h7.5M21 12a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="size-12 text-blue-700 z-100"
        data-slot="icon"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m12.75 15 3-3m0 0-3-3m3 3h-7.5M21 12a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`FunnelPreview renders msg when no children 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-items-center"
  >
    <div
      class="bg-cover bg-center h-[919px] w-[453px] pt-[96px] pb-[103px] pl-[26px] pr-[20px] z-1"
      style="background-image: url(/smartphone-cropped.png);"
    >
      <div
        class="h-full w-full p-4"
        style="background-color: white;"
      >
        <div
          class="slick-slider slick-initialized"
        >
          <div
            class="slick-list"
          >
            <div
              class="slick-track"
              style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-hidden="false"
                class="slick-slide slick-active slick-current"
                data-index="0"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="flex justify-center items-center text-center w-full h-full mt-8"
                    style="width: 100%; display: inline-block;"
                    tabindex="-1"
                  >
                    <p
                      class="text-xl text-gray-500"
                    >
                      No funnel data to show. Load a JSON file to display a preview.
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="grid grid-cols-2 justify-items-center flex-none w-full text-center mt-4"
    >
      <svg
        aria-hidden="true"
        class="size-12 text-blue-700"
        data-slot="icon"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m11.25 9-3 3m0 0 3 3m-3-3h7.5M21 12a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="size-12 text-blue-700 z-100"
        data-slot="icon"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m12.75 15 3-3m0 0-3-3m3 3h-7.5M21 12a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </div>
  </div>
</div>
`;
